AWSCognitoAuth/AWSCognitoAuth.m

```diff
@@ -589,6 +590,7 @@ - (NSString *) urlEncode: (NSString *) stringToEncode {
 }
     
 - (BOOL)application:(UIApplication *)app openURL:(NSURL *)url options:(NSDictionary<UIApplicationOpenURLOptionsKey, id> *)options {
+    self.isHandlingRedirection = YES;
```

Comment:
I think this is a bit too high-level a place to be setting this flag: in cases where the `processURL:` method never reaches the point where it establishes an NSURLConnection, we would never clear the flag.

I'm going to accept the PR, and then I'll make a change for the next release to set this flag only if we reach the point where we actually establish the NSURLConnection to handle the auth token. I'll also verify my belief that `safariViewController:didCompleteInitialLoad:` will always be invoked *after* `application:openURL:options:` in redirect scenarios.

Reply:
Sounds good. I initially was going to put this flag where the NSURLConnection for the oauth2/token call. I'm pretty indifferent about it so feel free to move it there or however you see fit. The only real reason I put it where it is now is because the variable name is `isHandlingRedirection` and technically the `application:openURL:` is where that begins in this class.